Mathlib/LinearAlgebra/Basis.lean

```diff
@@ -1397,14 +1397,43 @@ end Induction
 with the basis elements. -/
 lemma Basis.mem_center_iff {A}
     [Semiring R] [NonUnitalNonAssocSemiring A]
-    [Module R A] [SMulCommClass R A A] [IsScalarTower R A A]
-    (b : Basis ι R A) {x : A} :
-    x ∈ Set.center A ↔ ∀ i, Commute (b i) x := by
+    [Module R A] [SMulCommClass R A A] [SMulCommClass R R A] [IsScalarTower R A A]
+    (b : Basis ι R A) {z : A} :
+    z ∈ Set.center A ↔ (∀ i, Commute (b i) z) ∧ ∀ i j,
+    z * (b i * b j) = (z * b i) * b j
+    ∧ (b i * z) * b j = b i * (z * b j)
+    ∧ (b i * b j) * z = b i * (b j * z) := by
   constructor
-  · intro h i; apply h
-  · intro h y
-    rw [← b.total_repr y, Finsupp.total_apply, Finsupp.sum, Finset.sum_mul, Finset.mul_sum]
-    simp_rw [mul_smul_comm, smul_mul_assoc, (h _).eq]
+  · intro h;
+    constructor
+    · intro i
+      apply (h.1 (b i)).symm
+    · intros
+      exact ⟨h.2 _ _, ⟨h.3 _ _, h.4 _ _⟩⟩
+  · intro h
+    rw [center, mem_setOf_eq]
+    constructor
+    · intro y
```

Comment:
Can you use the `case` tactic to name these goals?

Reply:
Done, thanks.